sotodlib/site_pipeline/update_preprocess_plots.py

```
@@ -0,0 +1,256 @@
+        default=2,
+        type=int
+    )
+    return parser
+
+def main(
+        configs: str,
+        query: Optional[str] = None, 
+        obs_id: Optional[str] = None, 
+        overwrite: bool = False,
+        min_ctime: Optional[int] = None,
+        max_ctime: Optional[int] = None,
+        update_delay: Optional[int] = None,
+        tags: Optional[str] = None,
+        planet_obs: bool = False,
+        verbosity: Optional[int] = None,
+ ):
+    configs, context = _get_preprocess_context(configs)
+    logger = sp_util.init_logger("preprocess", verbosity=verbosity)
+
+    if (min_ctime is None) and (update_delay is not None):
+        # If min_ctime is provided it will use that..
+        # Otherwise it will use update_delay to set min_ctime.
+        min_ctime = int(time.time()) - update_delay*86400
+
+    if obs_id is not None:
+        tot_query = f"obs_id=='{obs_id}'"
+    else:
+        tot_query = "and "
+        if min_ctime is not None:
+            tot_query += f"timestamp>={min_ctime} and "
+        if max_ctime is not None:
+            tot_query += f"timestamp<={max_ctime} and "
+        if query is not None:
+            tot_query += query + " and "
+        tot_query = tot_query[4:-4]
+        if tot_query=="":
+            tot_query="1"
+
+    if not(tags is None):
+        for i, tag in enumerate(tags):
+            tags[i] = tag.lower()
+            if '=' not in tag:
+                tags[i] += '=1'
+
+    if planet_obs:
+        obs_list = []
+        for tag in tags:
+            obs_list.extend(context.obsdb.query(tot_query, tags=[tag]))
+    else:
+        obs_list = context.obsdb.query(tot_query, tags=tags)
+    if len(obs_list)==0:
+        logger.warning(f"No observations returned from query: {query}")
+    run_list = []
+
+    if not os.path.exists(configs['archive']['index']):
+        #don't run if database doesn't exist
+        logger.warning(f"No database found at {configs['archive']['index']}.")
```

If you want this to break I think you should raise an exception instead of a warning here.